freud/box.pyx

```diff
@@ -921,6 +940,75 @@ cdef class Box:
                             "positional argument: L")
         return cls(Lx=L, Ly=L, Lz=0, xy=0, xz=0, yz=0, is2D=True)
 
+    @classmethod
+    def from_lattice_vectors(cls, lattice_vectors, dimensions: int = None):
```

Comment:
How is this different than `from_matrix`? `Box.from_matrix` makes a box from a matrix where the columns are the lattice vectors. This seems very similar to that.

Reply:
Not sure how this got past me but with this in mind this opens a whole new debate. `from_matrix` in freud basically uses lattice vectors as input, but this is not stated explicitly. Instead a link to hoomd docs is given which further complicates the problem. In hoomd `from_matrix` wants upper triangular matrix that is already a box like object as defined in hoomd docs. What we should do is to have freuds `from_matrix` take upper triangular matrix as input, such that its functionality is equivalent to hoomd's and add `from_lattice_vectors` which has same function as current `from_matrix`?

Reply:
I think it would be best to add to freud's documentation and explicitly state that the columns of the input to `from_matrix` are the lattice vectors of the box. Any other improvements to the documentation of `from_matrix` to make it less confusing would also be welcome.

Looking at the diff of the code between `from_lattice_vectors` and `from_matrix`, they do the exact same thing so I don't see a need to have both factory methods hanging around.

Reply:
Is the functionality of freud's `from_matrix` equivalent to hoomd's `from_matrix`?

Reply:
It appears that hoomd's `from_matrix` requires an upper triangular matrix, whereas freud's is more general and does not.